.grit/patterns/dict_get_instead_of_if_else_block.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Use dict.get with default instead of if-else
+---
+
+Join multiple with statements into a single one. Rule [SIM401](https://github.com/MartinThoma/flake8-simplify/issues/72) from [flake8-simplify](https://github.com/MartinThoma/flake8-simplify).
+
+Caveat: if either the key or the default value are functions, they will be called a different
+number of times in the generated code. We may need to enforce that `$key` and `$default` don't
+call any function.
+
+```grit
+engine marzano(0.1)
+language python
+
+// NOTE: if $key or $default call functions with side effects, this transform is not safe
```

Comment:
Let's handle that in the code. You can use a where clause.

Reply:
Done, I have added:

```grit
 where {
    !$key <: contains call(),
    !$default <: contains call(),
}
```

Of course, in Python it is tricky to 100% enforce this, as an operator could be overloaded with anything.